src/test/java/core/basesyntax/service/impl/FileWriterImplTest.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.service.impl;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exeption.WrongFileFormat;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("FileWriterImpl Test")
+class FileWriterImplTest {
+    private static FileWriterImpl fileWriterE;
+    private static final String VALID_DATA_PATH = "src/test/resources/ValidInputTestResult1.csv";
+    private static final String INVALID_DATA_PATH =
+            "src/test/resources/incorrectPath/ValidOutputTestResult1.csv";
+    private static final String EMPTY_DATA_PATH = "";
+
+    @BeforeAll
+    static void beforeAll() {
+        fileWriterE = new FileWriterImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        new File(VALID_DATA_PATH).delete();
+    }
+
+    @DisplayName("Check writing to file in correct path")
+    @Test
+    void writeDataToFile_correctPath_ok() {
+        List<String> expected = List.of(
+                "banana,152",
+                System.lineSeparator(),
+                "apple,90");
+        fileWriterE.writeDataToFile(VALID_DATA_PATH, expected);
+        expected = List.of("fruit,quantity",
+                "banana,152",
+                "apple,90");
```

Comment:
It's better not to reassign the variables.
```suggestion
        List<String> dataToWrite = List.of(
                "banana,152",
                System.lineSeparator(),
                "apple,90");
        fileWriterE.writeDataToFile(VALID_DATA_PATH, dataToWrite);
        List<String> expected = List.of("fruit,quantity",
                "banana,152",
                "apple,90");
```